src/components/ChatRoomPage/ChatItem.tsx

```diff
@@ -0,0 +1,69 @@
+import styled from "styled-components";
+import { Chat, User } from "../../common/interface";
+import { ProfileImage } from "../../common/StyledComponent";
+
+interface ChatItemProps {
+  chat: Chat;
+  isCurUser: boolean;
+  sender: User;
+}
+
+const ChatItem = ({ chat, isCurUser, sender }: ChatItemProps) => {
+  const time = String(new Date(chat.date).getHours()).padStart(2, "0");
+  const minute = String(new Date(chat.date).getMinutes()).padStart(2, "0");
+
+  return (
+    <Wrapper isCurUser={isCurUser}>
+      {isCurUser ? (
+        <>
+          <ChatWrapper>
+            {time}:{minute}
+            <ChatBalloon isCurUser={true}>{chat.text}</ChatBalloon>
+          </ChatWrapper>
+        </>
+      ) : (
+        <>
+          <ProfileImage src={sender.profileImage} />
+          <ContentWrapper>
+            {sender.name}
+            <ChatWrapper>
+              <ChatBalloon isCurUser={false}>{chat.text}</ChatBalloon>
+              {time}:{minute}
+            </ChatWrapper>
+          </ContentWrapper>
+        </>
+      )}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div`
+  display: flex;
+  gap: 10px;
+  justify-content: ${({ isCurUser }: { isCurUser: boolean }) =>
+    isCurUser ? "flex-end" : "flex-start"};
+  width: 100%;
+`;
+const ContentWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 5px;
+  font-family: 'GmarketSansMedium';
+`;
+const ChatWrapper = styled.div`
+  display: flex;
+  align-items: flex-end;
+  gap: 5px;
+  font-family: 'GmarketSansMedium';
+
+`;
+const ChatBalloon = styled.div`
+  background-color: ${({ isCurUser }: { isCurUser: boolean }) =>
+    isCurUser ? "yellow" : "white"};
+  padding: 10px;
+  border-radius: 5px;
+  word-break: break-all;
+  
+`;
+
+export default ChatItem;
```

Comment:
현재 input submit을 할 때 변경되지 않은 ChatItem들까지 렌더링 되고 있어서 ChatItem에 React.memo를 사용해서 렌더링 최적화를 해주면 좋을 것 같습니다. 
[React.memo](https://ui.toast.com/weekly-pick/ko_20190731)와 관련된 글 입니다! 

```suggestion
export default React.memo(ChatItem);
```

Reply:
오오..! React.memo를 한 번도 써본 적은 없는 것 같은데 알려주셔서 감사합니다!!:)